evm/test/TrimmedAmount.t.sol

```diff
@@ -156,4 +156,112 @@ contract TrimmingTest is Test {
 
         assertEq(expectedRoundTrip, amountRoundTrip);
     }
+
+    // invariant: forall (x: uint256, y: uint8, z: uint8),
+    //            (x <= type(uint64).max, y <= z)
+    //                    => (x.trim(x, 8).untrim(y) == x)
+    function testFuzz_trimIsLeftInverse(
+        uint256 amount,
+        uint8 fromDecimals,
+        uint8 toDecimals
+    ) public {
+        // this is guaranteed by trimming
+        uint256 amt = bound(amount, 1, type(uint64).max);
+        vm.assume(fromDecimals <= 18);
```

Comment:
Could we fuzz with more decimals than this? Think there are some tokens with 24 decimals for example

Reply:
yea i tested with a few more decimals. should we add an error if the decimal amount is too large (say 100?). the division becomes very numerically unstable, creating underflow issues.

Reply:
I can't imagine that would ever be a realistic scenario right as so many other protocols would break if a token appeared with a crazy number of decimals! So I think going up to 50 is more than enough coverage, thanks